### PR TITLE
New Resource: `azurerm_system_center_virtual_machine_manager_availability_set`

### PR DESCRIPTION
**Pull request by neil-yechenwei**
--- PASS: TestAccSystemCenterVirtualMachineManagerAvailabilitySetSequential/scvmmAvailabilitySet/basic (269.23s)
--- PASS: TestAccSystemCenterVirtualMachineManagerAvailabilitySetSequential/scvmmAvailabilitySet/requiresImport (275.72s)
--- PASS: TestAccSystemCenterVirtualMachineManagerAvailabilitySetSequential/scvmmAvailabilitySet/complete (253.74s)
--- PASS: TestAccSystemCenterVirtualMachineManagerAvailabilitySetSequential/scvmmAvailabilitySet/update (327.84s)

References:
[Azure doc](https://learn.microsoft.com/en-us/azure/azure-arc/system-center-virtual-machine-manager/overview)
[REST API specs](https://github.com/Azure/azure-rest-api-specs/blob/01e99457ccf5613a95d5b2960d31a12f84018863/specification/scvmm/resource-manager/Microsoft.ScVmm/stable/2023-10-07/scvmm.json#L1490)

### CODE REVIEW

File: internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_availability_set_resource.go [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+			state := SystemCenterVirtualMachineManagerAvailabilitySetModel{}
+			if model := resp.Model; model != nil {
+				state.Name = id.AvailabilitySetName
+				state.ResourceGroupName = id.ResourceGroupName
+				state.Location = location.Normalize(model.Location)
+				state.CustomLocationId = pointer.From(model.ExtendedLocation.Name)
+				state.Tags = pointer.From(model.Tags)
+
+				scvmmServerId, err := vmmservers.ParseVMmServerID(pointer.From(model.Properties.VMmServerId))
```
Comment (ms-zhenhua) [inline internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_availability_set_resource.go:151]:
```suggestion
				scvmmServerId, err := vmmservers.ParseVMmServerIDInsensitively(pointer.From(model.Properties.VMmServerId))
```

File: internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_availability_set_resource.go [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+				return err
+			}
+
+			var model SystemCenterVirtualMachineManagerAvailabilitySetModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			parameters := availabilitysets.ResourcePatch{
+				Tags: pointer.To(model.Tags),
```
Comment (ms-zhenhua) [inline internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_availability_set_resource.go:180]: Though currently only `tags` can be updated. other updatable properties may be added in future. Shall we still use `if d.HasChange("tags")` here?

File: internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_availability_set_resource_test.go [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+		},
+		data.ImportStep(),
+	})
+}
+
+func testAccSystemCenterVirtualMachineManagerAvailabilitySet_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_system_center_virtual_machine_manager_availability_set", "test")
+	r := SystemCenterVirtualMachineManagerAvailabilitySetResource{}
+
+	data.ResourceSequentialTest(t, r, []acceptance.TestStep{
```
Comment (ms-zhenhua) [inline internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_availability_set_resource_test.go:85]: To test `add`, `update` and `removal` of updatable properties, shall we test with `basic -> complete -> update -> basic`?

**Review by ms-zhenhua — CHANGES_REQUESTED** [review]
Hi @neil-yechenwei,
Thanks for this PR - I've taken a look through and left some comments inline. If we can fix those up, this should be good to go 👍

File: internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_availability_set_resource_test.go (hunk 3 above)
Comment (neil-yechenwei) [reply internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_availability_set_resource_test.go:85]: Updated

File: internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_availability_set_resource.go (hunk 2 above)
Comment (neil-yechenwei) [reply internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_availability_set_resource.go:180]: Updated

File: internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_availability_set_resource.go (hunk 1 above)
Comment (neil-yechenwei) [reply internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_availability_set_resource.go:151]: Updated

**Comment by neil-yechenwei** [comment]
@ms-zhenhua , thanks for the comments. I updated PR. Please take another look. Thanks.

**Review by ms-zhenhua — APPROVED** [review]

**Review by katbyte — APPROVED** [review]
LGTM 🌵

**Pull request merged**